maestrowf/utils.py

```diff
@@ -41,9 +41,26 @@
 LOGGER = logging.getLogger(__name__)
 
 
+def get_duration(time_delta):
+    """
+    Covert durations to HH:MM:SS format.
+
+    :params time_delta: A time difference in datatime format.
+    :returns: A formatted string in HH:MM:SS
+    """
+    duration = time_delta.total_seconds()
+    days = int(duration / 86400)
+    hours = int((duration % 86400) / 3600)
+    minutes = int((duration % 86400 % 3600) / 60)
+    seconds = int((duration % 86400 % 3600) % 60)
+
+    return "{:d} days {:02d}h:{:02d}m:{:02d}s" \
```

Comment:
I'm actually wondering if the format should be like -  2d:3h:3m:3s that makes it more consistent and easier to parse since everything is 2 digits and a letter.

Reply:
You know, that’s a good point. I had originally had that and then changed it. But the silly thing is that I don’t necessarily have to limit the days to 2 digits like the rest. I’ll make a new commit that changes the format.

Reply:
You know, that’s a good point. I had originally had that and then changed it. But the silly thing is that I don’t necessarily have to limit the days to 2 digits like the rest. I’ll make a new commit that changes the format.